crates/cheatnet/src/cheatcodes/deploy.rs

```diff
@@ -77,6 +78,20 @@ impl CheatnetState {
                     ))))
                 }),
             CallContractOutput::Panic { panic_data } => {
+                for panic_felt in &panic_data {
+                    let err_str = as_cairo_short_string(panic_felt).unwrap();
+                    for invalid_calldata_msg in [
+                        "Failed to deserialize param #",
+                        "Input too long for arguments",
+                    ] {
+                        if err_str.contains(invalid_calldata_msg) {
+                            return Err(CheatcodeError::Unrecoverable(EnhancedHintError::from(
+                                CustomHint(Box::from(err_str)),
```

Comment:
We only pass a part of an error here though, what if `panic_data.len() > 1`? We should convert all felts to shortstrings and concatenate them imo